app/queries/gobierto_people/people_with_activities_query.rb

```diff
@@ -0,0 +1,106 @@
+# frozen_string_literal: true
+
+module GobiertoPeople
+  class PeopleWithActivitiesQuery < RowchartItemsQuery
+
+    def initialize(params = {})
+      @site = params[:site]
+      @conditions = params[:conditions].symbolize_keys
+      @relation = (params[:relation] || model.all).where(id: people_with_activities.map(&:id)).left_outer_joins(events_association)
+      append_query_conditions(@conditions) if @conditions
+      @limit = params[:limit] || DEFAULT_LIMIT
+    end
+
+    private
+
+    def events_association
+      :attending_events
+    end
+
+    ASSOCIATIONS = {
+      events:
+      {
+        association_name: :attending_events,
+        start_date: "#{GobiertoCalendars::Event.table_name}.starts_at >= :start_date",
+        end_date: "#{GobiertoCalendars::Event.table_name}.ends_at <= :end_date"
+      },
+      trips:
+      {
+        association_name: :trips,
+        start_date: "#{GobiertoPeople::Trip.table_name}.start_date >= :start_date",
+        end_date: "#{GobiertoPeople::Trip.table_name}.end_date <= :end_date"
+      },
+      invitations:
+      {
+        association_name: :invitations,
+        start_date: "#{GobiertoPeople::Invitation.table_name}.start_date >= :start_date",
+        end_date: "#{GobiertoPeople::Invitation.table_name}.end_date <= :end_date"
+      },
+      gifts:
+      {
+        association_name: :received_gifts,
+        start_date: "#{GobiertoPeople::Gift.table_name}.date >= :start_date",
+        end_date: "#{GobiertoPeople::Gift.table_name}.date <= :end_date"
+      }
+    }.freeze
+
+    def people_with_activities
+      GobiertoPeople::Person.find_by_sql(activities_sql)
+    end
+
+    def model
+      Person
+    end
+
+    def append_query_conditions(conditions)
+      if conditions[:department_id]
+        @relation = Department.filter_department_people(
+          conditions.slice(:department_id, :start_date, :end_date)
+          .merge(people_relation: relation)
+        )
+      end
+
+      if conditions[:interest_group_id]
+        append_condition(:interest_group_id, conditions[:interest_group_id])
+      end
+
+      if conditions[:start_date]
+        append_condition(:starts_at, conditions[:start_date], ">=", true)
+      end
+
+      if conditions[:end_date]
+        append_condition(:ends_at, conditions[:end_date], "<=", true)
+      end
+    end
+
+    def table_name
+      model.table_name
+    end
+
+    def activities_sql
+      cte = ASSOCIATIONS.keys.map do |association_model|
```

Comment:
What does cte stands for? The variable name could be improved a bit 😝

Reply:
It stands for `common_table_expression`. I've changed `cte` with it, it's much more expressive ;)